### ISSUE TITLE
Feature Request: Processing Framework Integration

### ISSUE THREAD

**Issue by jakimowb** [SMByC/AcATaMa#7]
AcATaMa is a great plugin! It would be nice if some of its tool can be used in the QGIS Processing Framework, as this allows to include them into the QGIS Model builder. 

The first ones that come to my mind are:

[ ] Simple Random Sampling
[ ] Stratified Random Sampling
[ ] Accuracy Assessment

**Comment by XavierCLL** [comment]
Hi Jakimowb, thanks, that is a really nice idea in the future

**Comment by jakimowb** [comment]
It would be also nice if potentially long-running algorithms could use the QGIS Task Manager. This way they can run in parallel, do not block the QGIS GUI and might use the "Cancel process" flag: https://docs.qgis.org/3.16/en/docs/pyqgis_developer_cookbook/tasks.html 
I know, parallelization of python code is complicated when using PyQt/QGIS. But in this moment I am waiting for the stratified sampling getting fininished to sample > 240000 points, which takes quite some minutes in which I'd like to use QGIS for other tasks as well.

**Comment by XavierCLL** [comment]
Yes, I'm aware of this, I want too I'll try (be patient) to parallelize the sampling process it is quite complicated due to several checking functions to guarantee all the constraints. I created issue #8 for that

**Comment by Josi-Nic** [comment]
First of all, AcATaMa is a really nice plugin!
I have one question and two suggestions:
Is it possible to create the accuracy assessment report with a fully labeled point layer? I have created and labeled the points with the AcATaMa plugin, but ran into an issue. One of my two classes was falsely recognised as "no data value". I fixed this in the attribute table using the field calculator (but keeping the field structure as it was, I only altered the values in the columns). But sadly, after that fix I was not able to calculate the accuracy assessment any more with the AcATaMa plugin. Is there a way to do that? If not, it would be really nice to add that feature. Because there are some reference resources that can't be loaded into QGIS (like eg. the historical google earth images from google earth pro). And in that case it would anyway be nice if one could create a point layer and label the points without using the plugin (but in the required format), but calculate the accuracy assessment with it.
And here my two suggestions: 
It would be really nice, if during the labeling one could skip to the next point using the arrow keys and if one could assign classes using the digit keys. Please let me know if this is already possible somehow, but I just didn't find out how.
